compiler/aarch64/codegen/OMRMemoryReference.cpp

```diff
@@ -852,6 +858,20 @@ uint8_t *OMR::ARM64::MemoryReference::generateBinaryEncoding(TR::Instruction *cu
                      }
                   }
                }
+            else if (isImm7OffsetGPRInstruction(enc))
+               {
+               uint32_t opc = ((enc >> 30) & 3); /* 32bit: 00, 64bit: 10 */
+               uint32_t shifted = displacement >> ((opc >> 1) + 2);
```

Comment:
What about adding a `TR_ASSERT()` for making sure `displacement` is a multiple of 4 or 8 ?

Reply:
I agree. Added assertion for `displacement` alignment.